ingest/annotations.py

```diff
@@ -98,12 +109,19 @@ def create_data_frame(self):
         )[0]
         # dtype of object allows mixed dtypes in columns, including numeric dtypes
         # coerce group annotations that pandas detects as non-object types to type string
-        for annotation, annot_type in columns:
-            if (
-                annot_type == 'group'
-                and self.file[annotation].dtypes[annot_type] != 'O'
-            ):
-                self.file[annotation] = self.file[annotation].astype(str)
+        if 'group' in self.annot_types:
+            group_columns = self.file.xs(
+                "group", axis=1, level=1, drop_level=False
+            ).columns.tolist()
+            try:
+                # coerce group annotations to type string
+                self.file[group_columns] = self.file[group_columns].astype(str)
+            except Exception as e:
+                self.error_logger.error(
+                    "Unable to coerce group annotation to string type",
```

Comment:
It's odd that our auto-formatter permits inconsistent quotes -- double here, single in line 112.

Reply:
Sadly Black would enforce using all double quotes. Black is generally not configurable. This particular choice is problematic enough that Black has a --skip-string-normalization as an option, which we use. So Black is allowing inconsistency unless we want enforcement of all double quotes.

Reply:
Could we use single-quotes here for consistency, in that case, or would they trigger a warning / be reformatted?

Reply:
Sure, I've gone through the file and tried to fix all the quoting inconsistencies `22cec53`. It's a pity there's no way to have Black enforce single quotes.